Classes/Plugin/Collection.php

```diff
@@ -129,12 +129,10 @@ protected function showCollectionList() {
                 ),
                 Helper::whereExpression('tx_dlf_collections')
             )
-            ->orderBy($orderBy)
-            ->execute();
-
-        $allResults = $result->fetchAll();
+            ->orderBy($orderBy);
 
-        $count = count($allResults);
+        $result = $queryBuilder->execute();
+        $count = $queryBuilder->count('uid')->execute()->fetchColumn(0);
         $content = '';
         if ($count == 1 && empty($this->conf['dont_show_single'])) {
             $resArray = $allResults[0];
```

Comment:
This won't work, since there is no `$allResults` anymore.

Reply:
You're right. I'll send a patch.

Unfortunately we have another bug above in the query. The orderBy with FIELD()-function seems not to be possible with doctrine :-(

Reply:
Do you provide a separate pull request for the `FIELD()` bug or should I keep this one open?

Reply:
I'll provide a separate patch. Actually, I'm not yet sure how to fix it.